### PR TITLE
Messaging service

### PR DESCRIPTION
**Pull request by tristan957**
Abstract GroupMeSevice further into an impl of MessagingService in preparation for a potential re-org of the code.

### CODE REVIEW

File: src/handlers/HeyHandler.ts [hunk 1]
```diff
@@ -11,9 +13,9 @@ export default class HeyHandler implements Handler {
     regexps: HeyHandler.regexps,
   };
 
-  constructor(@inject(GroupMeService) private readonly groupMeService: GroupMeService) {}
+  constructor(@inject("MessagingService") private readonly messagingService: MessagingService) {}
```
Comment (JosephMart) [inline src/handlers/HeyHandler.ts]: Why is this a string now?

**Review by JosephMart — APPROVED** [review]
Not sure I understand the string in the injection part enough but LGTM :)

File: src/handlers/HeyHandler.ts (hunk 1 above)
Comment (tristan957) [reply src/handlers/HeyHandler.ts]: Because it is an interface. Classes have prototypes for getting injection tokens, but interfaces do not.

Comment (tristan957) [reply src/handlers/HeyHandler.ts]: TS being held back by JS is what I think it boils down to.

**Comment by tristan957** [comment]
#10 should be merged into this PR before merging to master.

**Review by JosephMart — APPROVED** [review]
assuming it runs :P, LGTM

**Pull request merged**